### PR TITLE
For proxy

### PR DESCRIPTION
**Pull request by dcillera**
Envoy-openssl with some commits needed for Maistra Proxy
PR created to test CPAAS procedure in Proxy.

Signed-off-by: Dario CILLERAI [dcillera@redhat.com](mailto:dcillera@redhat.com)

### CODE REVIEW

File: bssl-compat/BUILD [hunk 1]
```diff
@@ -15,7 +15,12 @@ cmake(
     visibility = ["//visibility:public"],
     generate_crosstool_file = False,
     tags = ["requires-network"],
-    env = { "GOCACHE" : "/tmp" },
+    env = { "GOCACHE" : "/tmp",
+            "CMAKE_C_COMPILER" : "clang",
+            "CMAKE_CXX_COMPILER" : "clang++",
```
Comment (tedjpoole) [inline bssl-compat/BUILD:20]: When invoking cmake, bazel already passes `CC` & `CXX` in the environment, which is enough for cmake to do its thing. Therefore, `CMAKE_C_COMPILER` & `CMAKE_CXX_COMPILER` can be removed.

File: bssl-compat/CMakeLists.txt
Comment (tedjpoole) [inline bssl-compat/CMakeLists.txt:1]: Also remove `source/SSL_was_key_usage_invalid.cc`

**Review by tedjpoole — CHANGES_REQUESTED** [review]
A few changes & tidy ups required

File: bssl-compat/BUILD (hunk 1 above)
Comment (dcillera) [reply bssl-compat/BUILD:20]: This resulted necessary when building proxy (CMake couldn't find compilers) . However I 'll try again to confirm this.

File: bssl-compat/patch/include/openssl/ssl.h.sh [hunk 2]
```diff
@@ -193,11 +193,9 @@ uncomment.sh "$1" --comment -h \
   --uncomment-macro SSL_TICKET_KEY_NAME_LEN \
   --uncomment-enum ssl_verify_result_t \
   --uncomment-func-decl SSL_CTX_set_custom_verify \
-  --uncomment-func-decl SSL_CTX_set_reverify_on_resume \
   --uncomment-func-decl SSL_CTX_set_private_key_method \
   --uncomment-func-decl SSL_send_fatal_alert \
   --uncomment-func-decl SSL_alert_desc_string_long \
-  --uncomment-func-decl SSL_set_enforce_rsa_key_usage \
   --uncomment-func-decl SSL_was_key_usage_invalid \
```
[the remark this answers is not in the document]
Comment (dcillera) [reply bssl-compat/patch/include/openssl/ssl.h.sh:199]: SSL_was_key_usage_invalid is also called in envoy-openssl/source/extensions/transport_sockets/tls/context_impl.cc

File: bssl-compat/CMakeLists.txt
Comment (dcillera) [reply bssl-compat/CMakeLists.txt:1]: SSL_was_key_usage_invalid is also called in envoy-openssl/source/extensions/transport_sockets/tls/context_impl.cc, however I'll try it

File: bssl-compat/BUILD (hunk 1 above)
Comment (tedjpoole) [reply bssl-compat/BUILD:20]: Without `CMAKE_C_COMPILER` or `CMAKE_CXX_COMPILER` in `bssl-compat/BUILD`, I confirmed the following builds still work for me locally:
- Your `dcillera/envoy-openssl[proxy-envoy-openssl]` branch, in the `quay.io/maistra-dev/maistra-builder:2.6` builder.
- Your `dcillera/proxy[for-proxy]` branch, in the upstream builder.

File: bssl-compat/CMakeLists.txt
Comment (tedjpoole) [reply bssl-compat/CMakeLists.txt:1]: I left a comment on `context_impl.cc` to add `#if 0` around the `SSL_was_key_usage_invalid()` call

File: source/extensions/transport_sockets/tls/context_impl.cc [hunk 3]
```diff
@@ -569,14 +571,14 @@ void ContextImpl::logHandshake(SSL* ssl) const {
 #error "Delete preprocessor check below; no longer needed"
 #endif
 
-#if BORINGSSL_API_VERSION >= 18
+#if 0
```
Comment (tedjpoole) [inline source/extensions/transport_sockets/tls/context_impl.cc:574]: Could you reinstate the original upstream code, and then add an *additional* `#if 0` with a comment, just around the `SSL_was_key_usage_invalid()` call. That way, the intent is clearer, and we are modifying upstream code less by only making a small addition of the `#if 0`.

**Review by tedjpoole — CHANGES_REQUESTED** [review]

**Review by tedjpoole — APPROVED** [review]

**Pull request merged**